File: code_samples/lua/testing_fixtures.lua
-- Testing: Fixtures
-- AI/ML Training Sample

Fixtures = {}
Fixtures.__ index = Fixtures

function Fixtures:new()
    local obj = {
        data = ""
    }
    setmetatable(obj, self)
    return obj
end

function Fixtures:process(input)
    self.data = input
end

function Fixtures:getData()
    return self.data
end

function Fixtures:validate()
    return #self.data > 0
end

-- Example usage
local instance = Fixtures:new()
instance:process("example")
print("Data: " .. instance:getData())
print("Valid: " .. tostring(instance:validate()))

File: code_samples/elixir/testing_fixtures.ex
# Testing: Fixtures
# AI/ML Training Sample

defmodule Fixtures do
  defstruct data: ""
  
  def new(), do: %Fixtures{}
  
  def process(%Fixtures{} = struct, input) do
    %{struct | data: input}
  end
  
  def get_data(%Fixtures{data: data}), do: data
  
  def validate(%Fixtures{data: data}) do
    String.length(data) > 0
  end
end

# Example usage
instance = Fixtures.new()
updated = Fixtures.process(instance, "example")
IO.puts("Data: " <> Fixtures.get_data(updated))
IO.puts("Valid: " <> to_string(Fixtures.validate(updated)))
